Firestore/core/src/firebase/firestore/util/executor_libdispatch.mm

```diff
@@ -246,28 +260,33 @@ void MarkDone() {
   // guaranteed to outlive the executor, and it's possible for work to be
   // invoked by libdispatch after the executor is destroyed. Executor only
   // stores an observer pointer to the operation.
+  TimeSlot* time_slot = nullptr;
+  TimeSlotId time_slot_id = 0;
+  RunSynchronized(this, [this, delay, &operation, &time_slot, &time_slot_id] {
+    time_slot_id = time_slot_counter_++;
+    time_slot = new TimeSlot{this, delay, std::move(operation), time_slot_id};
+    schedule_[time_slot_id] = time_slot;
+  });
 
-  auto const time_slot = new TimeSlot{this, delay, std::move(operation)};
   dispatch_after_f(delay_ns, dispatch_queue(), time_slot,
                    TimeSlot::InvokedByLibdispatch);
-  RunSynchronized(this, [this, time_slot] { schedule_.push_back(time_slot); });
-  return DelayedOperation{[this, time_slot] {
+
+  return DelayedOperation{[this, time_slot_id] {
     // `time_slot` might be destroyed by the time cancellation function runs.
     // Therefore, don't access any methods on `time_slot`, only use it as
```

Comment:
The comment is outdated now. Maybe something like:
```
// `time_slot` might have been destroyed by the time cancellation function runs,
// in which case it's guaranteed to be already removed from the schedule. If the
// `time_slot_id` refers to a slot that has been removed, the call to
// `RemoveFromSchedule` will be a no-op.
```

Reply:
Rephrased.